internal/controller/runner_controller.go

```diff
@@ -0,0 +1,275 @@
+// SPDX-License-Identifier: MIT
+
+package controller
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/cloudbase/garm/client/instances"
+	"github.com/cloudbase/garm/params"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	garmoperatorv1alpha1 "github.com/mercedes-benz/garm-operator/api/v1alpha1"
+	garmClient "github.com/mercedes-benz/garm-operator/pkg/client"
+	"github.com/mercedes-benz/garm-operator/pkg/client/key"
+)
+
+// RunnerReconciler reconciles a Runner object
+type RunnerReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+
+	BaseURL  string
+	Username string
+	Password string
+}
+
+//+kubebuilder:rbac:groups=garm-operator.mercedes-benz.com,namespace=xxxxx,resources=runners,verbs=get;list;watch;create;update;patch;delete
```

Comment:
as in the generated `yaml` already mentioned, we should remote the `create` param here as we do not create runners

```suggestion
//+kubebuilder:rbac:groups=garm-operator.mercedes-benz.com,namespace=xxxxx,resources=runners,verbs=get;list;watch;update;patch;delete
```